packages/ag-charts-community/src/chart/series/cartesian/areaUtil.ts

```diff
@@ -30,400 +32,133 @@ export type AreaPathDatum = {
+        readonly fill: string;
+    };
+}
 
-                    if (bottomPoints.length > 0) {
-                        fill.path.lineTo(
-                            bottomPoints[bottomPoints.length - 1].x,
-                            bottomPoints[bottomPoints.length - 1].y
-                        );
-                    }
+export interface AreaSeriesNodeDataContext
+    extends CartesianSeriesNodeDataContext<MarkerSelectionDatum, LabelSelectionDatum> {
+    fillData: AreaPathDatum;
+    strokeData: AreaPathDatum;
+}
 
-                    fill.path.closePath();
-                    fill.checkPathDirty();
-                },
-            });
+function renderPartialArea(pairData: PathPoint[], ratios: Partial<Record<PathPointChange, number>>, path: Path) {
+    const { path: areaPath } = path;
+    let previousTo: PathPoint['to'];
+    for (const data of pairData) {
+        const ratio = ratios[data.change];
+        if (ratio == null) continue;
+
+        const { from, to } = data;
+        if (from == null || to == null) continue;
+
+        const x = from.x + (to.x - from.x) * ratio;
+        const y = from.y + (to.y - from.y) * ratio;
+
+        if (data.moveTo === false) {
+            areaPath.lineTo(x, y);
+        } else if (data.moveTo === true || !previousTo) {
+            areaPath.moveTo(x, y);
+        } else if (previousTo) {
+            const moveToRatio = data.moveTo === 'in' ? ratio : 1 - ratio;
+            const midPointX = previousTo.x + (x - previousTo.x) * moveToRatio;
+            const midPointY = previousTo.y + (y - previousTo.y) * moveToRatio;
+            areaPath.lineTo(midPointX, midPointY);
+            areaPath.moveTo(x, y);
         }
-
-        markerSelections[seriesIdx]?.each((marker, datum) => {
-            const delay = seriesRect?.width ? (datum.point.x / seriesRect.width) * duration : 0;
-            const format = formatter ? ctx.callbackCache.call(formatter as any, getFormatterParams(datum)) : null;
-
-            ctx.animationManager.animate({
-                id: `${seriesId}_empty-update-ready_${marker.id}`,
-                groupId: seriesId,
-                ...animationOptions,
-                to: format?.size ?? datum.point?.size ?? 0,
-                delay,
-                duration: markerDuration,
-                onUpdate(size) {
-                    marker.size = size;
-                },
-            });
-        });
-
-        labelSelections[seriesIdx].each((label, datum) => {
-            ctx.animationManager.animate({
-                id: `${seriesId}_empty-update-ready_${label.id}`,
-                groupId: seriesId,
-                from: 0,
-                to: 1,
-                delay: seriesRect?.width ? (datum.x / seriesRect.width) * duration : 0,
-                duration: markerDuration,
-                onUpdate(opacity) {
-                    label.opacity = opacity;
-                },
-            });
-        });
-    });
+        previousTo = to;
+    }
```

Comment:
This looks the same as the `render()` function inside `prepareLinePathAnimation()` in `lineUtils`. Is it possible to reuse this at all?

Reply:
Well, that got turned into `renderPartialLine()` in this diff, but still looks the same as this

Reply:
Very good shout - they are identical, I think they were different at some intermediate stage but they turned out to just be a general 'animated-path' rendering - I've renamed and moved to `pathUtil.ts`.